src/components/split-button/split-button-rtl.test.tsx

```diff
@@ -0,0 +1,912 @@
+    render(
+      <SplitButton text="Main" data-element="bar" data-role="baz">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+
+    expect(toggle).toHaveAttribute("data-element", "toggle-button");
+  });
+
+  it("should render with correct data attributes on the additional button element container", async () => {
+    const user = userEvent.setup();
+    render(
+      <SplitButton text="Main" data-element="bar" data-role="baz">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+    await user.hover(toggle);
+    const additionalButtons = await screen.findByRole("list");
+
+    expect(additionalButtons).toHaveAttribute(
+      "data-element",
+      "additional-buttons"
+    );
+  });
+
+  it("should render with custom id when 'id' prop is passed", () => {
+    render(
+      <SplitButton text="Main" id="custom-id">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    expect(screen.getByRole("button", { name: "Main" })).toHaveAttribute(
+      "id",
+      "custom-id"
+    );
+  });
+
+  it("should render with custom class when 'className' prop is passed", () => {
+    render(
+      <SplitButton text="Main" className="custom-class">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    expect(screen.getByRole("button", { name: "Main" })).toHaveClass(
+      "custom-class"
+    );
+  });
+
+  it("should render with disabled state when 'disabled' prop is passed", () => {
+    render(
+      <SplitButton text="Main" disabled>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    expect(screen.getByRole("button", { name: "Main" })).toBeDisabled();
+    expect(screen.getByRole("button", { name: "Show more" })).toBeDisabled();
+  });
+
+  it("should render non-Carbon Button children", async () => {
+    const user = userEvent.setup();
+    const spanElement = <span>span-element</span>;
+    render(<SplitButton text="Main">{spanElement}</SplitButton>);
+
+    await user.click(screen.getByRole("button", { name: "Show more" }));
+    expect(await screen.findByText("span-element")).toBeInTheDocument();
+  });
+
+  it("should render with the correct styles when 'buttonType' prop is 'primary' and not disabled", () => {
+    render(
+      <SplitButton text="Main" buttonType="primary">
+        {singleButton()}
+      </SplitButton>
+    );
+
+    expect(screen.getByRole("button", { name: "Show more" })).toHaveStyle({
+      position: "relative",
+    });
+  });
+
+  it("should render with the correct styles when 'buttonType' prop is 'primary' and disabled", () => {
+    render(
+      <SplitButton text="Main" buttonType="primary" disabled>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    expect(screen.getByRole("button", { name: "Show more" })).not.toHaveStyle({
+      position: "relative",
+    });
+  });
```

Comment:
**suggestion**: I'm considering if these two tests are necessary here, given the position isn't being asserted along with offset properties like `top`, `bottom`, etc. What do you think about removing these?

Reply:
Needed for coverage, I don't think asserting the rest of the styles matters here, they're tested in the PW tests

Reply:
I do need these for coverage